python/triton/language/semantic.py

```diff
@@ -436,6 +436,9 @@ def not_equal(input: tl.tensor,
 
 
 def arange(start: int, end: int, builder: ir.builder) -> tl.tensor:
+    if not isinstance(start, int) or not isinstance(end, int):
+        raise ValueError("arange's arguments must be of type tl.constexpr")
```

Comment:
If both `start` and `end` are int, they are not an instance of `tl.constexpr`.

Do you want to say `ValueError("arange's arguments must be of int")`?

Reply:
Hmm, I know what you meant. `start` and `end` are `int` at compilation time...I think you are right but the annotation here is a bit confusing. @ptillet Any thoughts?

Reply:
At the semantic analysis level, they are indeed `int`. The frontend extracts the value out of the constexpr (and the ir::value* handle out of the tensors) before passing them to the semantic analysis. https://github.com/openai/triton/blob/master/python/triton/language/core.py#L636-L647

Reply:
Ha, I see. Then should this check be placed in `core.py`? Just like what you did in the `zeros` function:

https://github.com/openai/triton/blob/4bf509889bae68ef20668ff6edadbb293c7e431f/python/triton/language/core.py#L661

Reply:
There might be other places we want to restrict the arguments as constexpr as well:

https://github.com/openai/triton/blob/4bf509889bae68ef20668ff6edadbb293c7e431f/python/triton/language/core.py#L626

https://github.com/openai/triton/blob/4bf509889bae68ef20668ff6edadbb293c7e431f/python/triton/language/core.py#L614

I suppose we can have a discussion about on which level these error messages should prompt, as we are going to keep the current frontend. 

Currently, I do see error messages such as, I'm not sure if they provide with enough information for users:

> TypeError: create_get_program_id(): incompatible function arguments. The following argument types are supported:
>     1. (self: triton._C.libtriton.triton.ir.builder, arg0: int) -> triton._C.libtriton.triton.ir.value